### PR TITLE
feat: adds redirects and basic custom 404 page

### LINKED ISSUE #698: docs: rewrites and custom 404
### Is your feature request related to a problem? Please describe.

Would be nice with some rewrites and a custom 404 page

https://beta.create.t3.gg/en Currently 404s to an ugly page, would be nice to rewrite it to either / or /en/introduction

other pages, e.g.  https://beta.create.t3.gg/en/page-that-dont-exist should go to a custom 404 page

### Describe the solution you'd like to see

See above

### Describe alternate solutions

-

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by PiotrekPKP**
Closes #698 

## ✅ Checklist

- [x] I have followed every step in the [contributing guide](https://github.com/t3-oss/create-t3-app/blob/main/CONTRIBUTING.md) (updated 2022-10-06).
- [x] The PR title follows the convention we established [conventional-commit](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] I performed a functional test on my final commit

---

## Changelog

- Added redirect from `/en` to `/`
- Added basic custom 404 page with the landing page layout

---

## Screenshots

image

💯

### CODE REVIEW

File: www/src/pages/en.astro [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+const redirectUrl = "/";
```
Comment (juliusmarminge) [inline www/src/pages/en.astro:2]: We can do this in the vercel.json:

https://vercel.com/docs/project-configuration#project-configuration/redirects

File: www/src/pages/404.astro [hunk 2]
```diff
@@ -0,0 +1,60 @@
+---
+import LandingPage from "../layouts/landingPage.astro";
+import Button from "../components/landingPage/button.astro";
+---
+
+<LandingPage>
```
Comment (juliusmarminge) [inline www/src/pages/404.astro:6]: There is some weird thing with the height of the page (at least on my phone) See: https://create-t3-90d36hrg1-t3-oss.vercel.app/fff

Comment (juliusmarminge) [inline www/src/pages/404.astro:6]: Not sure where we pass in the flag to hide the themeToggler, but it should prob be hidden here (behavior is weird)

File: www/src/pages/en.astro (hunk 1 above)
Comment (PiotrekPKP) [reply www/src/pages/en.astro:2]: I wanted to separate the deployment from the actual site but I guess it won't matter here, we're not changing Vercel for anything haha
The con of doing it in Vercel's file is that you can't really have this behavior in the dev env (does it matter at all?)

File: www/src/pages/404.astro (hunk 2 above)
Comment (PiotrekPKP) [reply www/src/pages/404.astro:6]: Yeah, I'm trying to fix the `isLanding` things everywhere

File: www/src/pages/en.astro (hunk 1 above)
Comment (juliusmarminge) [reply www/src/pages/en.astro:2]: It's fine I think?

**Comment by juliusmarminge** [comment]
Throw a `h-screen` on there maybe?

CleanShot 2022-11-06 at 15 30 07

**Comment by PiotrekPKP** [comment]
Oh yeah, I thought you were talking about the white thingy in the bottom, haven't seen that

**Review by asrvd — COMMENTED** [review]
looks great, the vercel footer needs to be fixed though, adding a `h-full` on the hero div may fix it, or i can make one more pr making improvements once this gets merged

**Comment by PiotrekPKP** [comment]
I can do it @asrvd

**Comment by PiotrekPKP** [comment]
Actually h-full did not work and I had to calc the height of the hero section but it did it so yeah

**Review by asrvd — APPROVED** [review]

**Comment by nexxeln** [comment]
Conflicts :(

**Comment by nexxeln** [comment]
Conflicts :(

**Comment by PiotrekPKP** [comment]
I'll hopefully resolve them today, most likely in 12 hours tho

**Comment by PiotrekPKP** [comment]
Should be ok :)

**Review by juliusmarminge — APPROVED** [review]

**Pull request merged**